### PR TITLE
Add a View menu, improve grid plotting

### PR DESCRIPTION
**Pull request by bendudson**
Updates the gui options, and triggers a replotting of the grid.

Some plotting settings are in the Preferences dialog. The View menu and preferences dialog are now kept in sync.

Fixed the plot_grid keep_limits functionality, so that the xlim and ylim settings are preserved when replotting.

- [x] Updated manual
- [x] Updated `doc/whats-new.md` with a summary of the changes

### CODE REVIEW

**Comment by bendudson** [comment]
The main purpose of all this was to make grid generation issues easier to diagnose visually, and to be able to calculate and plot a `penalty_mask` variable for imposing penalization boundaries in BOUT++ models.
240123-hypnotoad

File: doc/whats-new.md [hunk 1]
```diff
@@ -1,6 +1,20 @@
...
 ===============
 
+### Bug fixes
+
+
+### New features
+
+- Radial grid line construction can recover from failure and generate
+  a rough grid to help visual inspection when option
+  `follow_perpendicular_recover` is set to True
```
Comment (johnomotani) [inline doc/whats-new.md:11]:
```suggestion
  `follow_perpendicular_recover` is set to True (#175)
```

File: doc/whats-new.md [hunk 2]
```diff
@@ -1,6 +1,20 @@
...
+- A `View` menu enables the grid plot to be customised, with cell edges, corners,
+  grid lines and other components.
```
Comment (johnomotani) [inline doc/whats-new.md:13]:
```suggestion
  grid lines and other components (#176).
```

File: doc/whats-new.md [hunk 3]
```diff
@@ -1,6 +1,20 @@
...
+- `penalty_mask` is calculated and written to the grid file, based on intersection
+  of the grid with the wall. This enables immersed boundary conditions.
+- Wall coordinates are written to output grid as `closed_wall_R` and `closed_wall_Z`
```
Comment (johnomotani) [inline doc/whats-new.md:16]:
```suggestion
- Wall coordinates are written to output grid as `closed_wall_R` and `closed_wall_Z`
  (#176)
```

File: integrated_tests/utils.py [hunk 4]
```diff
@@ -17,6 +17,10 @@
     "hypnotoad_inputs_yaml",
     "Python_version",
     "module_versions",
+    # Variables that have been added
```
Comment (johnomotani) [inline integrated_tests/utils.py:20]:
```suggestion
    # Variables that have been added. These entries can be removed if/when the
    # expected output is re-generated.
```

**Review by johnomotani — COMMENTED** [review]
Thanks @bendudson, looks good to me.

One remaining problem - there is (I think) a bug in Qt.py's `--convert` feature, which means that on this branch the gui is not working unless PySide2 is installed. The `hypnotoad_mainWindow.py` file (generated by pyside2-uic) now contains
```python
from PySide2.QtCore import *
```
But the `--convert` feature only does [this](https://github.com/mottosso/Qt.py/blob/183c18791c70a27355149b84844d5da17c238b0e/Qt.py#L1720-L1722):
```python
        line = line.replace("from PySide2 import", "from Qt import QtCompat,")
        line = line.replace("QtWidgets.QApplication.translate",
                            "QtCompat.translate")
```

I guess this should be raised as an issue on Qt.py, but the short-term fix would be to edit `hypnotoad_mainWindow.py` to use the Qt.py equivalent of the PySide2 import (whatever that is). Or we could just require PySide2, but that's probably more work than sticking with Qt.py.

**Pull request merged**